### PR TITLE
chore: Delete root Makefile

### PR DESCRIPTION
**Pull request by erezrokah**
#### Summary

Fixes https://github.com/cloudquery/cloudquery/issues/3913 that was experienced by a Discord user trying to contribute.
Since running all targets in the make file can be slow, this PR does it twice per day.

I also removed 2 targets:
1. `install-tools` was failing. I think it needs to be per module
2. `pre-commit` didn't do anything as we need to run go linter per module with the relevant config

Happy to open issues for ⬆️ if we feel like we need them

### CODE REVIEW

**Comment by erezrokah** [comment]
Blocked by https://github.com/cloudquery/plugin-sdk/pull/373

File: .github/workflows/monorepo_makefile_test.yml
Comment (erezrokah) [inline .github/workflows/monorepo_makefile_test.yml:7]: This is here for testing purposes. I'll remove it before merging

**Comment by yevgenypats** [comment]
Can we remove the root makefile at all? is it at all used by anyone? I only use a plugin specific makefile?

**Review by yevgenypats — APPROVED** [review]

File: Makefile
Comment (disq) [inline Makefile]: I know this is for CI only, but it doesn't work on mac unless you do `grep -E` at the start. It also seems to find some targets (from targets that invoke other targets) multiple times.

```
grep -E '^[^#[:space:].].*$$' Makefile | sed 's/\(.*\):/"\1",/' | tr -d '\n' | sed 's/,$$//g'
"gen-mocks","test","install-tools","install-hooks","gen-docs","lint","gen-code","gen", gen-code gen-mocks gen-docs
```

**Pull request merged**
